pattern_import_export/models/ir_exports.py

```diff
@@ -63,47 +76,122 @@ def generate_pattern(self):
     @api.multi
     def _get_data_to_export(self, records):
         """
-        Iterator who built data dict record by record
+        Iterator who built data dict record by record.
+        This function could be recursive in case of sub-pattern
         @param records: recordset
         @return: dict
         """
         self.ensure_one()
-        headers = self._get_header()
         for record in records:
-            data = {}
-            fields_to_export = []
-            column_headers = {}
-            for export_line in self.export_fields:
-                nb_column = 1
-                if export_line.is_many2x and export_line.select_tab_id:
-                    field_name = export_line.select_tab_id.field_id.name
-                    field = export_line.name + "/" + field_name
-                    if export_line.is_many2many:
-                        nb_column = export_line.number_occurence
-                else:
-                    field = export_line.name
-                for _i in range(0, nb_column):
-                    fields_to_export.append(field)
-                column_headers.update({field: nb_column})
-            # The data-structure returned by export_data is different
-            # that the one used to export.
-            # export_data(...) return a dict with a keys named 'datas'
-            # and it contains a list of list.
-            # Each list item is a line (for M2M) but for the export,
-            # we want to display these lines as column.
-            # So we have to convert
-            res = record.export_data(fields_to_export, raw_data=False)
-            for target_ind, record_data in enumerate(res.get("datas", []), start=1):
-                for header, value in zip(headers, record_data):
-                    to_update = True
-                    if COLUMN_M2M_SEPARATOR in header:
-                        to_update = False
-                        _name, ind = header.split(COLUMN_M2M_SEPARATOR, 1)
-                        if ind.isdigit() and target_ind == int(ind):
-                            to_update = True
-                    if to_update and header not in data:
-                        data.update({header: value})
-            yield data
+            yield self._get_data_to_export_by_record(record)
+
+    @api.multi
+    @tools.ormcache("self", "record")
```

Comment:
I do not think it's a good idea. ORM cache is keep between transaction durint the whole life of the worker.
So if I do an export it will keep in memory the result and if my record change and I redo an export I will have the old data.
Also exporting a lot of data mean keeping them in memory until we restart odoo, so I can consume a lot of memory